hs_rest_api2/tests/test_resource_permission.py

```diff
@@ -0,0 +1,179 @@
+            self.view_group,
+            PrivilegeCodes.VIEW
+        )
+
+        # URL for the permission endpoint
+        self.url = "/hsapi2/resource/{}/permission/json/".format(self.res.short_id)
+
+    def test_get_permission_owner(self):
+        # Test owner permission
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+        self.assertEqual(response_json['permission'], 'owner')
+
+    def test_get_permission_edit(self):
+        # Test edit permission
+        self.client.logout()
+        self.client.login(username='edituser', password='edituser')
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+        self.assertEqual(response_json['permission'], 'edit')
+
+    def test_get_permission_view(self):
+        # Test view permission
+        self.client.logout()
+        self.client.login(username='viewuser', password='viewuser')
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+        self.assertEqual(response_json['permission'], 'view')
+
+    def test_get_permission_none(self):
+        # Test no permission
+        self.client.logout()
+        self.client.login(username='nopermuser', password='nopermuser')
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+        self.assertEqual(response_json['permission'], 'none')
+
+    def test_get_permission_nonexistent_resource(self):
+        # Test with non-existent resource
+        nonexistent_url = "/hsapi2/resource/abc123/permission/json/"
+        response = self.client.get(nonexistent_url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+        self.assertEqual(response_json['permission'], 'none')
+
+    def test_get_permission_anonymous_user(self):
+        # Test no permission
+        self.client.logout()
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+        self.assertEqual(response_json['permission'], 'none')
+
+    def test_get_permission_group_edit(self):
+        # Test edit permission through group membership
+        self.client.logout()
+        self.client.login(username='groupedituser', password='groupedituser')
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+        self.assertEqual(response_json['permission'], 'edit')
```

Comment:
The way you have it, `groupedituser` should actually have `PrivilegeCodes.VIEW` not "edit"

Reply:
Okay but look at these 2 lines:
https://github.com/hydroshare/hydroshare/pull/5914/files#diff-cdfdfb6597c83b22e79e50113fe977e6421ee717da1f241c8a15ecba3906be5cR67
https://github.com/hydroshare/hydroshare/pull/5914/files#diff-cdfdfb6597c83b22e79e50113fe977e6421ee717da1f241c8a15ecba3906be5cR72
the comments don't make sense

Reply:
Do you have any suggestion how to improve those comments? Or should I just delete those comments?

Reply:
sorry I missed this question before
I will take a look on Monday